### PR TITLE
Restructure quick screen

### PR DESCRIPTION
**Pull request by mdrlzy**

### CODE REVIEW

File: app/src/main/java/dev/arkbuilders/rate/presentation/quick/QuickViewModel.kt [hunk 1]
```diff
@@ -72,34 +73,46 @@ class QuickViewModel(
...
+                        frequent = frequent,
+                        topResults = topResults
+                    )
+                }
+            }.launchIn(viewModelScope)
+
+            val all = currencyRepo.getCurrencyNameUnsafe()
+            val frequent = calcFrequentCurrUseCase()
+                .map { currencyRepo.nameByCodeUnsafe(it) }
+            val topResults = frequent + (all - frequent)
```
Comment (hieuwu) [inline app/src/main/java/dev/arkbuilders/rate/presentation/quick/QuickViewModel.kt:104]: Nice trick to move the frequent list to the top. 😂 Perhaps consider having 2 use cases like `GetTopResult` and other use case to encapsulate these logic

File: app/src/main/java/dev/arkbuilders/rate/presentation/quick/QuickViewModel.kt [hunk 2]
```diff
@@ -56,6 +56,7 @@ class QuickViewModel(
     private val quickRepo: QuickRepo,
     private val prefs: Prefs,
     private val convertUseCase: ConvertWithRateUseCase,
+    private val calcFrequentCurrUseCase: CalcFrequentCurrUseCase,
```
Comment (hieuwu) [inline app/src/main/java/dev/arkbuilders/rate/presentation/quick/QuickViewModel.kt:59]: It looks like `assetsRepo`, `prefs`, and `convertUseCase` are not used in this ViewModel. Can we remove them?

File: app/src/main/java/dev/arkbuilders/rate/presentation/quick/QuickViewModel.kt [hunk 3]
```diff
@@ -72,34 +73,46 @@ class QuickViewModel(
...
+                val topResults = frequent + (currencies - frequent)
+                reduce {
+                    state.copy(
+                        frequent = frequent,
+                        topResults = topResults
+                    )
+                }
+            }.launchIn(viewModelScope)
+
+            val all = currencyRepo.getCurrencyNameUnsafe()
```
Comment (hieuwu) [inline app/src/main/java/dev/arkbuilders/rate/presentation/quick/QuickViewModel.kt:101]: As `currencyRepo.getCurrencyNameUnsafe()` returns the same result, I think we can lift it out of the for loop to me in outer level to reduce calling times to data source
```suggestion
           val allCurrencies = currencyRepo.getCurrencyNameUnsafe()
            calcFrequentCurrUseCase.flow().drop(1).onEach {
                val frequent = calcFrequentCurrUseCase.invoke()
                    .map { currencyRepo.nameByCodeUnsafe(it) }
                val topResults = frequent + (allCurrencies - frequent)
                reduce {
                    state.copy(
                        frequent = frequent,
                        topResults = topResults
                    )
                }
            }.launchIn(viewModelScope)

            val all = allCurrencies // replace all with allCurrencies
```

Comment (mdrlzy) [reply app/src/main/java/dev/arkbuilders/rate/presentation/quick/QuickViewModel.kt]: fixed

File: app/src/main/java/dev/arkbuilders/rate/presentation/quick/QuickViewModel.kt (hunk 2 above)
Comment (mdrlzy) [reply app/src/main/java/dev/arkbuilders/rate/presentation/quick/QuickViewModel.kt:65]: fixed

File: app/src/main/java/dev/arkbuilders/rate/presentation/quick/QuickViewModel.kt (hunk 1 above)
Comment (mdrlzy) [reply app/src/main/java/dev/arkbuilders/rate/presentation/quick/QuickViewModel.kt]: It's a good idea to extract use case.

**Review by hieuwu — APPROVED** [review]
Looks good.

**Review by hieuwu — APPROVED** [review]

**Review by shubertm — APPROVED** [review]

**Pull request merged**